### DISCUSSION TITLE
Issue with classes with the same name

### DISCUSSION THREAD

**Discussion by frankcoutinho** [LemonAppDev/konsist#1022] [Report a Problem]
### Tell us what version of Konsist you use.

0.15.1

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

```
package nice

import com.lemonappdev.konsist.api.Konsist
import com.lemonappdev.konsist.api.ext.list.functions
import com.lemonappdev.konsist.api.ext.list.parameters
import com.lemonappdev.konsist.api.ext.list.types
import com.lemonappdev.konsist.api.ext.provider.hasAnnotationOf

annotation class Frank

@Frank
interface ContractA {
    fun method(arg: Foo)
    data class Foo(val bar: String)
}

@Frank
interface ContractB {
    fun method(arg: Foo)
    data class Foo(val nice: String)
}

@Frank
interface ContractC {
    fun method(arg: Foo)
    data class Foo(val other: String)
}


fun main() {
    val project = Konsist.scopeFromProject()
    project
        .interfaces()
        .filter { it.hasAnnotationOf<Frank>() }
        .functions()
        .parameters
        .types
        .forEach { println(it.asClassDeclaration()?.fullyQualifiedName) }
}
```

When you have classes with **the same name**.
The first class declaration is returned where the other classes are expected.

### Actual behaviour

Prints
```
nice.ContractA.Foo
nice.ContractA.Foo
nice.ContractA.Foo
```

### Expected behaviour

Prints
```
nice.ContractA.Foo
nice.ContractB.Foo
nice.ContractC.Foo
```

**Comment by nataliapeterwas** [comment]
Thank you for reporting the problem.
We have already created a ticket to fix this issue: https://lemonappdev.atlassian.net/browse/KON-621

**Reply by nataliapeterwas** [reply]
We will fix this issue in the next release: [PR](https://github.com/LemonAppDev/konsist/pull/1254)

**Comment by igorwojda** [comment]
Fixed in https://github.com/LemonAppDev/konsist/releases/tag/v0.16.1
